### ISSUE TITLE
application plugin build error

### ISSUE THREAD

**Issue by Rusteam** [python-poetry/poetry#7573]
- **Poetry version**:  Poetry (version 1.4.0)
- **Python version**:  Python 3.10.6
- **OS version and name**:  macOS 12.6
- **pyproject.toml**:  https://github.com/Rusteam/poetry-check-existing-plugin/blob/main/pyproject.toml

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

When I try to `poetry build` my application plugin i get the following error:
```

  CheckExistingPlugin.activate() takes 2 positional arguments but 3 were given

  at ~/opt/miniconda3/envs/poetry_env/lib/python3.10/site-packages/poetry/plugins/plugin_manager.py:71 in activate
       67│         self._plugins.append(plugin)
       68│ 
       69│     def activate(self, *args: Any, **kwargs: Any) -> None:
       70│         for plugin in self._plugins:
    →  71│             plugin.activate(*args, **kwargs)
       72│ 
       73│     def _load_plugin_entry_point(self, ep: metadata.EntryPoint) -> None:
       74│         logger.debug("Loading the %s plugin", ep.name)  # type: ignore[attr-defined]
       75│ 
```

When inspecting the source code, I see that `PluginManager.activate` is called with `poetry` and `io` as args in `./poetry/factory.py`, while the application plugin docs clearly say there's only one argument to `ApplicationPlugin.activate` [func](https://python-poetry.org/docs/plugins/#application-plugins):
```
        plugin_manager = PluginManager(Plugin.group, disable_plugins=disable_plugins)
        plugin_manager.load_plugins()
        poetry.set_plugin_manager(plugin_manager)
        plugin_manager.activate(poetry, io)
```

Is there an inconsistency then, or am I doing something wrong?

**Comment by radoering** [comment]
Works for me. (I can build the linked project.) `Plugin.group` in `./poetry/factory.py` is `"poetry.plugin"` (not `"poetry.application.plugin"`), so only Non-ApplicationPlugins are activated in this place.

**Comment by Rusteam** [comment]
Right, I've installed `poetry` with pip as it had to be in the requirements, but after uninstalling it and running `poetry build` with system poetry installation, the project was built fine. Should it be clarified in the plugin development docs?

Although without pip-installed poetry I can't get a development support from my IDE.
![image](https://user-images.githubusercontent.com/22130831/222052706-2464a5d1-0fee-430e-9e6d-06338dc91bd6.png)
